sql/catalyst/src/test/scala/org/apache/spark/sql/catalog/v2/expressions/TransformExtractorSuite.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalog.v2.expressions
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst
+import org.apache.spark.sql.types.DataType
+
+class TransformExtractorSuite extends SparkFunSuite {
```

Comment:
This suite and the companion objects also look  independent to `v2 SessionCatalog`. Can we have a separate PR?

Reply:
The extractors are needed to convert transforms in the session catalog. This suite validates that any `Transform` instance that is an identity transform can be matched using `IdentityTransform`, even if the instance does not use Spark's internal case class.

Using these, the code needed to convert a table's transforms is much smaller. So these are needed for this PR to work.

Reply:
I know. I mean a separate PR before this, @rdblue .

Reply:
I suggested two prior PRs to help this `V2SessionCatalog` PR.

Reply:
Opened https://github.com/apache/spark/pull/24812

Reply:
Actually, I agree with you in general to avoid mixing changes together. It's just that I've been told that's not what the Spark community does when I've been on the side of suggesting similar changes.